storage/remote/queue_manager.go

```diff
@@ -46,152 +46,134 @@ const (
 )
 
 type queueManagerMetrics struct {
```

Comment:
can we leave a comment saying any metrics added need to be both `newQueueManagerMetrics` and `unregister`

Reply:
I added a test to enforce that any metric registered is also unregistered, WDYT?